src/vision/shiny_future/barbara_appreciates_performance_tools.md

```diff
@@ -0,0 +1,85 @@
+
+She refactors her code to resolve this issue, and then re-checks traces. This time, the endpoint behaves as expected, but she's not seeing the wall clock time she expects; the trace shows that she's waiting on a network call to another service (also written in async Rust), and it's taking about 10x longer to reply than she would expect. She looks into the tracing libraries, and finds two useful features:
+
+1. She can annotate code with extra information that appears on the traces.
+2. Every point in the code has access to a unique ID that can be passed to external services to let her correlate traces.
+
+Barbara adds annotations that let her know how many bytes she's sending to the external service; it's not unreasonable, so she's still confused. A bit of work with the service owner, and she can now get traces from the external service that have IDs she sends with a request in them. The tooling combines traces nicely, so that she can now trace across the network into the external service, and she realises that it's going down a slow code path because she set the wrong request parameters.
+
+With the extra insights from the external service's trace, she's able to fix up her code to run perfectly, and she gets the desired wins from async Rust. Plus, she's got a good arsenal of tooling to use when next she sees an unidentified problem.
+
+## 🤔 Frequently Asked Questions
+
+### What status quo story or stories are you retelling?
+
+* [Barbara compares some C++ code (and has a performance problem)](../status_quo/barbara_compares_some_cpp_code.md)
+* [Barbara wants Async Insights](../status_quo/barbara_wants_async_insights.md)
+
+### **What is [Alan] most excited about in this future? Is he disappointed by anything?**
+
+Alan is excited about how easy it is to find out when his projects don't work as expected. He's happy
+
+### **What is [Grace] most excited about in this future? Is she disappointed by anything?**
+
+Grace is happy because the performance tools give her all the low level insights she wants into her code, and shows her what's going on "behind the scenes" in the executor. As a C++ developer, she is also excited when she sees that Rust developers who see an issue with her services can give her useful information about exactly what they see her C++ doing - which she can correlate with her existing C++ performance tools via the unique ID.
+
+### **What is [Niklaus] most excited about in this future? Is he disappointed by anything?**
+
+Niklaus is content. The tooling tells him what he needs to know, and allows him to add interesting information to places where he'd otherwise be stuck trying to extract it via `println!()`. He's not entirely sure how to use some of the detailed information, but he can ignore it easily.
```

Comment:
Oh, that's what I did. But doesn't `println!()` cause some delays?

Reply:
Any debugging tool can cause some delays - the point of this "shiny future" story is to describe a case where the tools are amazing, and thus you can attach interesting information to traces, and filter them to see exactly what you need to know.